src/lms7002m/LMS7002M.cpp

```diff
@@ -1522,6 +1525,78 @@ int LMS7002M::SetFrequencySX(bool tx, float_type freq_Hz, SX_details* output)
     return 0;
 }
 
+/** @brief Sets SX frequency with Reference clock spur cancelation
+    @param Tx Rx/Tx module selection
+    @param freq_Hz desired frequency in Hz
+    @return 0-success, other-cannot deliver requested frequency
+*/
+int LMS7002M::SetFrequencySXWithSpurCancelation(bool tx, float_type freq_Hz, float_type BW)
+{
+    const float BWOffset = 2e6;
+    BW += BWOffset; //offset to avoid ref clock on BW edge
+    bool needCancelation = false;
+    float_type refClk = GetReferenceClk_SX(false);
+    int low = (freq_Hz-BW/2)/refClk;
+    int high = (freq_Hz+BW/2)/refClk;
+    if(low != high)
+        needCancelation = true;
+
+    int status;
+    float newFreq;
+    if(needCancelation)
+    {
+        newFreq = (int)(freq_Hz/refClk+0.5)*refClk;
+        TuneRxFilter(BW-BWOffset+2*abs(freq_Hz-newFreq));
```

Comment:
This spur cancellation was a special request and was needed only for receiver, it is not necessary for usual users.